### PR TITLE
Add Tutorials and Examples as external files.

### PR DESCRIPTION
**Pull request by jamesreaver**
Adding an example (or tutorial) now only requires adding the file name to fixtures.json and inserting the .c file into the Examples (Templates) folder.

The database is refreshed upon provisioning.

### CODE REVIEW

**Comment by ccadar** [comment]
Hi @giacomoguerci , can you move this last commit about provisioning to a different PR?  This PR is about adding tutorials and examples as external files.  Thanks!

**Comment by jamesreaver** [comment]
@ccadar The last commit ensures that any file added as an Example or Tutorial is embedded into Klee Web when provisioning is executed. This is why it follows this PR, it completes the addition of Examples and Tutorials. Should I proceed to create a separate PR anyway?

**Comment by ccadar** [comment]
I see, then it's fine to leave it here.  I'll take a look now, thanks.

File: src/klee_web/frontend/fixtures/Tutorials/maze.c [hunk 1]
```diff
@@ -10,8 +10,6 @@
 #include<stdio.h>
 #include<stdlib.h>
 
-#include <klee/klee.h>
```
Comment (ccadar) [inline src/klee_web/frontend/fixtures/Tutorials/maze.c]: Why did you remove this include?

Comment (jamesreaver) [inline src/klee_web/frontend/fixtures/Tutorials/maze.c]: The line triggers an error saying klee/klee.h is not found. I forgot to remove it when retrieving this example from the Klee website. I believe Klee Web already handles the inclusion of the klee header file, none of the other examples/tutorials have the line.
I could leave it in a comment explaining to the users that it would be needed outside of Klee Web.

**Comment by andronat** [comment]
@helicopter88 @giacomoguerci I see that circleci fails due to this error: `src/_fastmath.c:31:20: fatal error: Python.h: No such file or directory`. If I understand correctly this can be fixed by adding the `python-dev` package [here](https://github.com/klee/klee-web/blob/master/circle.yml#L9). Can you make 1 more commit adding it so we can see if will work?

**Comment by andronat** [comment]
Also, as a general remark, whatever resides under `src/klee_web/frontend/migrations` is not meant to me touched by hand. Every change in the migration scripts should be handled by the Django migration framework. I recommend (if you haven't already) to read the documentation about how those scripts should be handled [here](https://docs.djangoproject.com/en/1.9/topics/migrations/). This is important.

For this PR it is ok with the changes you did but in general it is not the proper way. The reason is that those files are meant to change the database during runtime. Every script is operating on-top of the previous one.

**Comment by jamesreaver** [comment]
I agree @andronat. The migrations files we found had already been modified by hand to load the examples; this commit only adds the tutorials. It would be good to refactor in the future to leave the migration files the way Django generates them.

File: src/klee_web/frontend/fixtures/Tutorials/maze.c (hunk 1 above)
Comment (ccadar) [inline src/klee_web/frontend/fixtures/Tutorials/maze.c]: I think the right thing would be to change the compilation line in the backend to add "-I <klee_source>/include" so that the right KLEE headers are included.  Could you do this?

**Comment by helicopter88** [comment]
In my branch, circle CI runs because of https://github.com/helicopter88/klee-web/commit/696a2a36dc195a88e3a1639aafcf09fa3e280149 ,but it needs the vault to be gone

File: src/klee_web/frontend/fixtures/Tutorials/maze.c (hunk 1 above)
Comment (jamesreaver) [inline src/klee_web/frontend/fixtures/Tutorials/maze.c]: I think it is what it's currently being done (line 41 of src/klee_web/worker/processor/klee_run.py). Here is how the klee comand is constructed:
`llvm_command = ['clang', '-I', '/src/klee/include', '-emit-llvm', '-c', '-g',  code_file, '-o', object_file]`
Is it okay? @ccadar

**Comment by andronat** [comment]
@helicopter88 if you have a branch that fixes circleci then why you don't make a PR with it? XD

**Comment by helicopter88** [comment]
Okay, by mistake I edited your commit, anyways, yes I can do that, but it has all the commits from the other PR

**Comment by andronat** [comment]
@helicopter88 I'm not sure that I understand what you say. It would be great if you create a branch from the master branch, then add those commits that fix circleci, make a separate PR with only those changes, and then all other PR should rebase.

**Comment by helicopter88** [comment]
Let's see if I can get it to work without the commits in the other PR

**Comment by andronat** [comment]
This needs a rebase.

**Comment by jamesreaver** [comment]
After we've merged the changes to support `--sym-stdin` I will get back to this PR (some tutorials will use that flag).
